watcher/watcher.go

```diff
@@ -44,4 +44,5 @@ func (w *Watcher[T]) Send(event T) bool {
 
 func (w *Watcher[T]) Close() {
 	w.eventCh.CloseAndDiscardQueued()
+	w.eventCh.Wait()
```

Comment:
is it safe to add `Wait` directly to `Close`? shouldn't we use different function `CloseAndWait` ?

Reply:
Yes this is safe and preferred. That way we can opt in where required without changing the existing code & behavior.